src/Servers/Kestrel/Transport.Quic/README.md

```diff
@@ -1,3 +1,167 @@
+
+    ```xml
+    <ItemGroup>
+        <PackageReference Include="Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic" Version="5.0.0-preview.3.20215.14" />
+    </ItemGroup>
+    ```
+
+    or you can add it via commandline
+
+    ```text
+    dotnet add package Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic --version "5.0.0-*"
+    ```
+
+4. Building msquic for schannel
+
+    5.0 preview3 and preview4 builds support h3-25, which needs to build a specific version of msquic. Checkout and build msquic via running the following.
+
+    ```text
+    cd ..
+    git clone --branch v0.9-draft-25 https://github.com/microsoft/msquic
+    cd msquic
+    ```
+
+    Next, build msquic targeting schannel and copy it to your application directory.
+
+    ```text
+    # requires Powershell Core to build MsQuic at the moment.
+    .\build.ps1 -Tls schannel
+    # Copy msquic to the app.
+    cp .\artifacts\windows\x64_Debug_schannel\msquic.dll ..\Http3App
+    ```
+
+    We are also planning a distribution story for msquic so you don't need to build it yourself.
+
+5. Add the following code to Program.cs
+
+    ```c#
+    namespace Http3App
+    {
+        public class Program
+        {
+            public static void Main(string[] args)
+            {
+                CreateHostBuilder(args).Build().Run();
+            }
+
+            public static IHostBuilder CreateHostBuilder(string[] args) =>
+                Host.CreateDefaultBuilder(args)
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        var cert = CertificateLoader.LoadFromStoreCert("Http3App", StoreName.My.ToString(), StoreLocation.CurrentUser, false);
+                        webBuilder.UseStartup<Startup>().UseQuic(options =>
+                        {
+                            options.Certificate = cert;
+                            options.Alpn = "h3-25";
+                        })
+                        .ConfigureKestrel((context, options) =>
+                        {
+                            options.EnableAltSvc = true;
+
+                            options.Listen(System.Net.IPAddress.Any, 5001, listenOptions =>
+                            {
+                                listenOptions.UseHttps(httpsOptions =>
+                                {
+                                    httpsOptions.ServerCertificate = cert;
+                                });
+                                listenOptions.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
+                            });
+                        });
+                    });
+        }
+    }
+    ```
+
+6. Run the app on commandline:
+
+    ```text
+    dotnet run
+    ```
+
+7. Launch Edge with switches to enable QUIC/
+
+    ```text
+    & "C:\Users\<user>\AppData\Local\Microsoft\Edge SxS\Application\msedge.exe" --enable-quic --quic-version=h3-25 --origin-to-force-quic-on=localhost:5001
```

Comment:
I'm assuming you need `-origin-to-force-quic-on` to force HTTP/3 on the first request? The `alt-svc` doesn't kick in until the second request?

Reply:
For localhost, Alt-Svc straight up doesn't work.

Reply:
Gotcha. Worth mentioning in the write-up. I was a little confused and assume others might be as well